curvefs/src/space/space_manager.h

```diff
@@ -0,0 +1,104 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef CURVEFS_SRC_SPACE_SPACE_MANAGER_H_
+#define CURVEFS_SRC_SPACE_SPACE_MANAGER_H_
+
+#include <bthread/mutex.h>
+
+#include <memory>
+#include <unordered_map>
+
+#include "curvefs/proto/mds.pb.h"
+#include "curvefs/src/space/bitmap_allocator.h"
+#include "curvefs/src/space/s3_allocator.h"
+#include "src/common/concurrent/rw_lock.h"
+
+namespace curvefs {
+namespace space {
+
+struct SpaceManagerOption {};
+
+class SpaceManager {
+ public:
+    SpaceManager() = default;
+    virtual ~SpaceManager() = default;
+
+    virtual SpaceStatusCode InitSpace(const mds::FsInfo& fsInfo) = 0;
+
+    virtual SpaceStatusCode UnInitSpace(uint32_t fsId, mds::FSType fsType) = 0;
+
+    virtual SpaceStatusCode StatSpace(uint32_t fsId, uint64_t* total,
+                                      uint64_t* available,
+                                      uint64_t* blkSize) = 0;
```

Comment:
Packing the three paramater "total, available, blkSize" in a struct will better.

Reply:
fix